core/src/main/java/io/opencensus/stats/Aggregate.java

```diff
@@ -0,0 +1,273 @@
+        Function<? super Range, T> p3,
+        Function<? super Mean, T> p4,
+        Function<? super StdDev, T> p5,
+        Function<? super Aggregate, T> defaultFunction) {
+      return p0.apply(this);
+    }
+  }
+
+  /** The count value of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class Count extends Aggregate {
+
+    Count() {
+    }
+
+    static Count create(long count) {
+      return new AutoValue_Aggregate_Count(count);
+    }
+
+    /**
+     * Returns the aggregated count.
+     *
+     * @return the aggregated count.
+     */
+    public abstract long get();
+
+    @Override
+    public final <T> T match(
+        Function<? super Sum, T> p0,
+        Function<? super Count, T> p1,
+        Function<? super Histogram, T> p2,
+        Function<? super Range, T> p3,
+        Function<? super Mean, T> p4,
+        Function<? super StdDev, T> p5,
+        Function<? super Aggregate, T> defaultFunction) {
+      return p1.apply(this);
+    }
+  }
+
+  /** The histogram of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class Histogram extends Aggregate {
+
+    Histogram() {
+    }
+
+    static Histogram create(long... bucketCounts) {
+      checkNotNull(bucketCounts, "bucket counts should not be null.");
+      List<Long> boxedBucketCounts = new ArrayList<Long>();
+      for (long bucketCount : bucketCounts) {
+        boxedBucketCounts.add(bucketCount);
+      }
+      return new AutoValue_Aggregate_Histogram(
+          Collections.unmodifiableList(boxedBucketCounts));
+    }
+
+    /**
+     * Returns the aggregated bucket counts. The returned list is immutable, trying to update it
+     * will throw an {@code UnsupportedOperationException}.
+     *
+     * @return the aggregated bucket counts.
+     */
+    public abstract List<Long> get();
+
+    @Override
+    public final <T> T match(
+        Function<? super Sum, T> p0,
+        Function<? super Count, T> p1,
+        Function<? super Histogram, T> p2,
+        Function<? super Range, T> p3,
+        Function<? super Mean, T> p4,
+        Function<? super StdDev, T> p5,
+        Function<? super Aggregate, T> defaultFunction) {
+      return p2.apply(this);
+    }
+  }
+
+  /** The range of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class Range extends Aggregate {
+
+    Range() {
+    }
+
+    static Range create(double min, double max) {
+      if (min != Double.POSITIVE_INFINITY && max != Double.NEGATIVE_INFINITY) {
+        checkArgument(min <= max, "max should be greater or equal to min.");
+      }
+      return new AutoValue_Aggregate_Range(min, max);
+    }
+
+    /**
+     * Returns the minimum of the population values.
+     *
+     * @return the minimum of the population values.
+     */
+    // TODO(songya): not v0.1, expose this method later.
+    public abstract double getMin();
```

Comment:
Should this TODO be removed, since the method is already public?  I think we will need it to be public in v0 for users to access the data in the `match` method.